core/templates/dev/head/domain/topic/TopicUpdateService.ts

```diff
@@ -415,98 +415,36 @@ oppia.factory('TopicUpdateService', [
           });
       },
 
-      /**
-       * Adds an additional story id to a topic and records the change
-       * in the undo/redo service.
-       */
-      addAdditionalStoryId: function(topic, storyId) {
-        var oldAdditionalStoryIdsList = angular.copy(
-          topic.getAdditionalStoryIds());
-        var newAdditionalStoryIdsList = angular.copy(oldAdditionalStoryIdsList);
-        newAdditionalStoryIdsList.push(storyId);
-        _applyTopicPropertyChange(
-          topic, TOPIC_PROPERTY_ADDITIONAL_STORY_IDS, newAdditionalStoryIdsList,
-          oldAdditionalStoryIdsList,
-          function(changeDict, topic) {
-            // Apply.
-            topic.addAdditionalStoryId(storyId);
-          }, function(changeDict, topic) {
-            // Undo.
-            topic.removeAdditionalStoryId(storyId);
-          });
-      },
-
       /**
        * Removes an additional story id from a topic and records the change
        * in the undo/redo service.
        */
-      removeAdditionalStoryId: function(topic, storyId) {
-        var oldAdditionalStoryIdsList = angular.copy(
-          topic.getAdditionalStoryIds());
-        var newAdditionalStoryIdsList = angular.copy(oldAdditionalStoryIdsList);
-        var index = newAdditionalStoryIdsList.indexOf(storyId);
-        if (index === -1) {
-          throw Error(
-            'Given story id is not present in additional stories of topic.');
-        }
-        newAdditionalStoryIdsList.splice(index, 1);
-        _applyTopicPropertyChange(
-          topic, TOPIC_PROPERTY_ADDITIONAL_STORY_IDS, newAdditionalStoryIdsList,
-          oldAdditionalStoryIdsList,
-          function(changeDict, topic) {
-            // Apply.
-            topic.removeAdditionalStoryId(storyId);
-          }, function(changeDict, topic) {
-            // Undo.
-            topic.addAdditionalStoryId(storyId);
-          });
-      },
-
-      /**
-       * Adds a canonical story id to a topic and records the change
-       * in the undo/redo service.
-       */
-      addCanonicalStoryId: function(topic, storyId) {
-        var oldCanonicalStoryIdsList = angular.copy(
-          topic.getCanonicalStoryIds());
-        var newCanonicalStoryIdsList = angular.copy(oldCanonicalStoryIdsList);
-        newCanonicalStoryIdsList.push(storyId);
-        _applyTopicPropertyChange(
-          topic, TOPIC_PROPERTY_CANONICAL_STORY_IDS, newCanonicalStoryIdsList,
-          oldCanonicalStoryIdsList,
-          function(changeDict, topic) {
-            // Apply.
-            topic.addCanonicalStoryId(storyId);
-          }, function(changeDict, topic) {
-            // Undo.
-            topic.removeCanonicalStoryId(storyId);
-          });
+      removeAdditionalStory: function(topic, storyId) {
```

Comment:
Bit odd that you're keeping removeAdditionalStory here but not addAdditionalStory. Any reason for the inconsistency?

Reply:
Yeah, that's because the add_story (both canonical and additional) functions have been shifted to the backend. The controller that creates the story automatically links it to the topic in the backend itself, since a story will always be linked to a topic.

Reply:
Ah, OK. I think it's worth adding a comment here to explain this (i.e. why this file contains no addAdditionalStory functions).

Reply:
Done, added it to the file overview at the top.